### PR TITLE
Connected mode always fetches the default quality profile for SQ6.6+

### LINKED ISSUE #464: Connected mode always fetches the default quality profile for SQ6.6+
### Description

The connected mode should fetch the quality profile specified for the project in SQ. However, the default quality profile is always retrieved when using SQ6.6+.

### Repro steps

The steps required to reproduce the problem:

1. Create a non-default QP in SQ
2. Setup a VS project in connected mode that uses the non-default QP

### Expected behavior

The correct quality profile is used to generate the ruleset used by the project.

### Actual behavior

The default quality profile is used.

### Known workarounds

None

### Related information

* SonarQube server version: 6.6+

The underlying issue is a breaking change on the SonarQube side - the parameter name to the _api/qualityprofiles/export_ API was changed, with support for the old parameter being dropped.

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #464

### CODE REVIEW

File: src/SonarQube.Client/Messages/RoslynExportProfileRequest.cs [hunk 1]
```diff
@@ -27,8 +27,14 @@ public class RoslynExportProfileRequest
         [JsonProperty("language")]
         public string LanguageKey { get; set; }
         [JsonProperty("name")]
-        public string QualityProfileName { get; set; }
+        public virtual string QualityProfileName { get; set; }
         [JsonProperty("exporterKey", Required = Required.Always)]
         public string ExporterKey => $"roslyn-{LanguageKey}";
     }
+
+    public class RoslynExportProfileRequestV66Plus : RoslynExportProfileRequest
```
Comment (Evangelink) [inline src/SonarQube.Client/Messages/RoslynExportProfileRequest.cs]: That's not ideal but this is the easiest fix I could think of.

Comment (duncanp-sonar) [reply src/SonarQube.Client/Messages/RoslynExportProfileRequest.cs]: Neat! Simpler than having to customise the JSON serializer.

File: src/SonarQube.Client/Services/SonarQubeService.cs [hunk 2]
```diff
@@ -228,7 +228,9 @@ public Uri GetProjectDashboardUrl(string projectKey)
         {
             EnsureIsConnected();
 
-            var request = new RoslynExportProfileRequest { QualityProfileName = qualityProfileName, LanguageKey = language.Key };
+            var request = this.serverVersion >= new Version(6, 6)
+                ? new RoslynExportProfileRequestV66Plus { QualityProfileName = qualityProfileName, LanguageKey = language.Key }
+                : new RoslynExportProfileRequest { QualityProfileName = qualityProfileName, LanguageKey = language.Key };
```
Comment (duncanp-sonar) [inline src/SonarQube.Client/Services/SonarQubeService.cs]: How hard would it be to add a test for this in _SonarQubeClientTests.cs_?

Comment (Evangelink) [reply src/SonarQube.Client/Services/SonarQubeService.cs]: Done

**Pull request merged**